src/PhpParser/Node/Value/ValueResolver.php

```diff
@@ -214,9 +214,6 @@ private function getConstExprEvaluator(): ConstExprEvaluator
         return $this->constExprEvaluator;
     }
 
-    /**
-     * @return mixed[]
-     */
```

Comment:
@dorrogeray it seems make PHPStan notice, seems due to register `ReturnAnnotationIncorrectNullableRector` and rectify CI running, could you verify? Thank you.

```
 ------------------------------------------------------------------------
  src/PhpParser/Node/Value/ValueResolver.php:217
 ------------------------------------------------------------------------
  - '#Method Rector\\Core\\PhpParser\\Node\\Value\\ValueResolver\:\:extractConstantArrayTypeValue\(\) return type has no value type specified in iterable type array#'
  - 
```

Reply:
the method can return `null`, so if changed, it should be `@return mixed[]|null` instead of removed.

Reply:
I did some investigation, and `ReturnAnnotationIncorrectNullableRector` rule correctly updates the phpdoc to `@return mixed[]|null`, however, its the `UnionTypesRector` which then removes the phpdoc, probably because it considers `@return mixed[]|null` to be equivalent with simple `?array` php return typehint. So its actually `UnionTypesRector` which needs that particular guard, or perhaps an option to not remove `mixed[]|null` in general, to avoid problems with phpstan which is configured to expect the array type. I will push a branch with my suggested fix shortly.

When running rector with just `ReturnAnnotationIncorrectNullableRector` rule:
![image](https://user-images.githubusercontent.com/4999006/163123905-b2265aaa-7f52-41e6-ac14-21516b95a1d6.png)
And then with just `UnionTypesRector` rule:
![image](https://user-images.githubusercontent.com/4999006/163123956-50e166ef-b252-4d2a-88a7-de8abfd00183.png)

Reply:
For multiple rules usage, you can create PR to https://github.com/rectorphp/rector-src/tree/main/tests/Issues

Reply:
Here is the fix: https://github.com/dorrogeray/rector-src/commit/75db5dd2c27ae7f938e5cf5beb86e5016d59da06
Can you cherrypick this commit or should I open a new PR instead of this one?

Reply:
separate PR with a new test case will be better, rollback existing code without changing to `@phpstan-return` as that seems still correct on https://github.com/dorrogeray/rector-src/commit/75db5dd2c27ae7f938e5cf5beb86e5016d59da06#diff-e8d39b18c808064651946f5daf0d52d8bdc6b67a6fbe165c466f5d6c24b9c894

Reply:
@samsonasik Here you go https://github.com/rectorphp/rector-src/pull/2067